addons/ofxAssimpModelLoader/src/ofxAssimpModelLoader.cpp

```diff
@@ -57,7 +57,7 @@ bool ofxAssimpModelLoader::loadModel(ofBuffer & buffer, bool optimize, const cha
 //------------------------------------------
 bool ofxAssimpModelLoader::load(const of::filesystem::path & fileName, int assimpOptimizeFlags){
 	file = ofToDataPath(fileName);
-	if (!of::filesystem::exists(file)) {
+	if (!std::filesystem::exists(file)) {
```

Comment:
eg here: no need to for these changes based on the discussion right?

Reply:
ah no, these changes are needed if the idea is not to import all of `std::filesystem` into `of::filesystem`! i think i understand the confusion:

the initial approach of this PR was to follow the current pattern and to import everything from `std::filesystem` into `of::filesystem` so that things like`of::filesystem::exists()` works (and historically, to support `boost::filesystem` and `experimental` and `std` on the same namespace, prior to C++17). but that's a lot of boilerplate, essentially manually `using` everyhing from `std::filesystem` except `path`, then implement a `path` in the missing slot. (it's not possible to "overwrite `path` in a complety imported `std::filesystem` namespace -- this was the semi-clever move to achieve it)

but after spending some time around to topic, it led to this latest, streamlined approach which does not do that: it embraces `std::filesystem::stuff()`, and provides an independent `of::filesystem::path` API that behaves exactly like `std::filesystem::path` plus 3 things: implicit narrow conversion, constructors for `std::filesystem::path` and implicit conversion to `std::filesystem::path`. 

(as mentionned above, github search reveals litteraly no usage of `of::filesystem::stuff()` outside of OF code)

this way, the only "custom" thing we have to support is (a) the API facade of `of::filesystem::path` that must match `std::filesystem::path`, and (b) the narrow conversion function. the rest is plain `std::filesystem`. moreover, in mac/linux, the actual implementation of `of::filesystem::path` can be `using std::filesystem::path`, so is it cost-free.

our methods take and produce `of::filesystem::path`, which can accept a `std::filesystem::path`, and will implicitely convert to `std::filesystem::path` (bascially extracting the underlying `path_`) when dropped into `std::filesystem::exists()` et al.

but important point to confirm: this is made with the idea that "proper hyigene" around `std::filesystem::path` will be enforced (i.e. all builtin addons and examples with be changed never to use `std::string` for a path). as such it will never be possible to do on windows `std::string a = std::filesystem::absolute(of::filesystem::path("../"));` but nobody is doing that right now. what people are doing is `std::string a = ofToDataPath("../");` and `void method(std::string path);`. this fixes it and today, ofxCv on windows will compile (i hope!!!). 

all future code would be `std::filesystem::path` cross-platform compatible with `auto a = ofToDataPath("../");` and and `void method(of::filesystem::path path);`; in that spirit we could also define:

```c++
namespace of::filesystem {
#if defined(win_32) and defined(OF_TOLERANT_NARROW_PATH_CONVERSION)
// custom class her
#else
using std::filesystem::path
#endif
}
```
so a future someone working in MSVC but using proper idioms (i.e never use a narrow std::string for paths) could plainly be working with `std::filesystem::path`.

(NB right now for testing purposes `ofConstants.h` is not modifield like that; the new class in active in mac/linux, but once things are decided, we can drop back to `std::filesystem::path` there)

functionally it could be done with your `ofFilePath` augmentation idea -- it's just about the name, but the name `of::filesystem::path` implies relations/functions that relate tightly to `std::filesystem::path`, and the potential substitution for actual `std::filesystem::path` makes it natural vs 3rd-party docs and tutorials.

Reply:
thanks @artificiel - I actually missed that those functions were `std::filesystem` ones not belonging to `std::filesystem::path` 

That all makes perfect sense and I think the idea of not importing the whole of std::filesystem as of::filesystem also makes sense. 

I'll leave it up to others to weigh in but this looks good to me.